### PR TITLE
2 svd2pac doesnt find referenced cluster if derivedfrom point to an item of array cluster

### PR DESCRIPTION
**Pull request by pellico**
By creating this pull request you agree to the terms in CONTRIBUTING.md.
https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md
--- DO NOT DELETE ANYTHING ABOVE THIS LINE ---

### Description 
While fixing this bug, code has been heavily restructured for easier support of _derived_from_ feature. _derived_from_ is supported also for registers. It has been enabled a feature in json in order to preserve the order of item tera template. Now the entity in generated code are in the same sequence as in xml.
No generated API.

### Related Issue
#2

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -365,25 +364,18 @@ an optional feature flag `tracing`. Enabling the feature provides the following
 additional functionalities:
 - an interface where register accesses can be piped though, enabling
   developers to log accesses to registers or even mock registers outright.
-  An implementation of that interface for unittests will be published in the near future.
+  An implementaion of that interface is provided by [`regmock-rs`](https://bitbucket.vih.infineon.com/projects/OXIDE/repos/regmock-rs/browse).
```
Comment (manczak-ifx) [inline README.md:367]: change link to the github repository

File: README.md [hunk 2]
```diff
@@ -365,25 +364,18 @@ an optional feature flag `tracing`. Enabling the feature provides the following
...
-The second feature flag that is added is `tracing_dummy`. By default tracing requires
-callbacks to be registered for logging. If no callbacks are registered the code will
-panic. If `tracing_dummy` is enabled the code will not panic and silently drop logs
-if no callbacks are registerd.
-
 ### Examples
 Below, some simple examples on how to use the tracing APIs are shown.
-For a more complete example 
+For a complete example of how to use the tracing features for
+e.g. unittesting see the documentation of [`regmock-rs`](https://bitbucket.vih.infineon.com/projects/OXIDE/repos/regmock-rs/browse).
```
Comment (manczak-ifx) [inline README.md:378]: change link to the github repository

**Review by manczak-ifx — CHANGES_REQUESTED** [review]
Please refer to the github project https://github.com/Infineon/regmock-rs

File: README.md (hunk 1 above)
Comment (pellico) [reply README.md]: Fixed

File: README.md (hunk 2 above)
Comment (pellico) [reply README.md]: Fixed

**Review by manczak-ifx — APPROVED** [review]
Build #12. Approved from CSS DSI.

**Review by andreasWallnerIFX — APPROVED** [review]

**Pull request merged**